src/build/buildController.ts

```diff
@@ -35,59 +36,62 @@ export class BuildController {
         this.buildExecutor = new BuildExecutor(context, platformInformation, environmentController, eventStream);
     }
 
-    public async build(uri: vscode.Uri, credential: Credential): Promise<void> {
-        if (!this.trySetAvailableFlag()) {
+    public async build(correlationId: string, uri: vscode.Uri, credential: Credential): Promise<void> {
+        let buildInput: BuildInput;
+        this.eventStream.post(new BuildTriggered(correlationId));
+        let start = Date.now();
+
+        try {
+            this.setAvailableFlag();

```

Comment:
can this line be included in the `try` block below?

Reply:
No, since the second try/catch has `finally` to reset the instance available flag. 

---
In reply to: [380571578](https://github.com/docascode/vscode-docs-build/pull/44#discussion_r380571578) [](ancestors = 380571578)